sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -299,19 +284,21 @@ class MapboxMap :
         transitionOptions?.let(style::setStyleTransition)
       },
       styleDataSourcesLoadedListener = { style ->
-        // TODO https://github.com/mapbox/mapbox-maps-android/issues/1371
-//        styleExtension.sources.forEach {
-//          it.bindTo(style)
-//        }
-//        styleExtension.layers.forEach { (layer, layerPosition) ->
-//          layer.bindTo(style, layerPosition)
-//        }
+        styleExtension.sources.forEach {
+          it.bindTo(style)
+        }
+        styleExtension.layers.forEach { (layer, layerPosition) ->
+          layer.bindTo(style, layerPosition)
+        }
       },
       styleDataSpritesLoadedListener = { style ->
-        // TODO https://github.com/mapbox/mapbox-maps-android/issues/1371
-//        styleExtension.images.forEach {
-//          it.bindTo(style)
-//        }
+        styleExtension.images.forEach {
+          it.bindTo(style)
```

Comment:
same question, could addition of an image affect addition of a models?

Reply:
Hmm, not sure - how could image affect models here? Do you mean loading images several times may throw an exception?

Reply:
I'm not sure what is behind ` it.bindTo(style)`, but probably exception would be thrown if there is an error, then models would not load, similarly to https://github.com/mapbox/mapbox-maps-android/pull/1378#discussion_r881411322

Reply:
Actually it would crash the app instead since the exception is thrown on the main thread. Models would not load but that's would not  be a problem anymore =)

Reply:
If an error is returned - exception will be thrown here https://github.com/mapbox/mapbox-maps-android/blob/main/extension-style/src/main/java/com/mapbox/maps/extension/style/image/ImageExtensionImpl.kt#L36 and the app will crash